static/js/apps/biomed_nl/app.tsx

```diff
@@ -102,55 +125,134 @@ function getSectionTrigger(title: string, opened: boolean): JSX.Element {
   );
 }
 
-function processApiResponse(response: BiomedNlApiResponse): DisplayedAnswer {
-  // TODO: format the markdown response
-  return {
-    answer: response.answer,
-    footnotes: response.footnotes,
-    debugInfo: response.debug,
-  };
-}
-
 const sampleQuestionToLink = (sampleQuestion: string): Link => ({
   id: sampleQuestion,
   title: sampleQuestion,
   url: `/experiments/biomed_nl#q=${encodeURIComponent(sampleQuestion)}`,
 });
 
+function constructFeedbackLink(response: BiomedNlApiResponse): string {
+  const queryField = `${FEEDBACK_QUERY_PARAM}${_.escape(response.query)}`;
+
+  const answerField = `${FEEDBACK_ANSWER_PARAM}${response.answer.substring(
+    0,
+    MAX_FORM_ANSWER_LENGTH
+  )}`;
+
+  const flattenedFootnotes = response.footnotes
+    .map(
+      (ref) =>
+        `${ref.refNum}-${ref.source}-${ref.isOutgoing ? "out" : "in"}-${
+          ref.prop
+        }-${ref.linkedType}`
+    )
+    .join(";");
+  const fullDebug = response.debug + "\n" + flattenedFootnotes;
+
+  const debugField = `${FEEDBACK_DEBUG_PARAM}${fullDebug.substring(
+    0,
+    MAX_FORM_DEBUG_LENGTH
+  )}`;
+  return `${FEEDBACK_FORM}${queryField}${answerField}${debugField}`;
+}
+
+function getFootnoteReferenceElementId(referenceNumber: number): string {
+  return `biomednl-ref-${referenceNumber}`;
+}
+
+function getReferenceBrowserElementId(reference: TripleReference): string {
+  return reference.isOutgoing
+    ? getOutArcRowElementId(reference.prop)
+    : getInArcSubsectionElementId(reference.linkedType, reference.prop);
+}
+
+function formatReferences(references: TripleReference[]): JSX.Element {
+  return (
+    <div
+      css={css`
+        margin-left: ${theme.spacing.lg}px;
+      `}
+    >
+      {references.map(
+        (reference: TripleReference, index: number): JSX.Element => {
+          const browserElementId = getReferenceBrowserElementId(reference);
+
+          const linkLabelItems = [reference.source, reference.prop];
+          if (!reference.isOutgoing) {
+            linkLabelItems.push(reference.linkedType);
+          }
+
+          return (
+            <p key={index}>
+              [{reference.refNum}]:{" "}
```

Comment:
nit: can use &nbsp

Reply:
done! thanks for the tip -- slowly learning!!